vortex-array/src/array/bool/compute/cast.rs

```diff
@@ -0,0 +1,20 @@
+use vortex_dtype::DType;
+use vortex_error::VortexResult;
+
+use crate::array::{BoolArray, BoolEncoding};
+use crate::compute::CastFn;
+use crate::{Array, IntoArray};
+
+impl CastFn<BoolArray> for BoolEncoding {
+    fn cast(&self, array: &BoolArray, dtype: &DType) -> VortexResult<Array> {
+        assert!(matches!(dtype, DType::Bool(_)));
+
+        // If the types are the same, return the array,
+        // otherwise set the array nullability as the dtype nullability.
+        if array.dtype() != dtype {
```

Comment:
This is already handled in the entry-point function, it should be (but probably isn't) documented in the CastFn docstring.